src/Aspire.Dashboard/Model/DashboardClient.cs

```diff
@@ -151,31 +150,19 @@ GrpcChannel CreateChannel()
 
             X509CertificateCollection GetFileCertificate()
             {
-                var filePath = configuration["ResourceServiceClient:ClientCertificate:FilePath"];
-                var password = configuration["ResourceServiceClient:ClientCertificate:Password"];
-
-                if (filePath is null or [])
-                {
-                    throw new InvalidOperationException("ResourceServiceClient:ClientCertificate:Source is \"File\", but no Certificate:FilePath is configured.");
-                }
+                var filePath = _dashboardOptions.ResourceServiceClient.ClientCertificates.FilePath!;
+                var password = _dashboardOptions.ResourceServiceClient.ClientCertificates.Password;
 
                 return [new X509Certificate2(filePath, password)];
             }
 
             X509CertificateCollection GetKeyStoreCertificate()
             {
-                var subject = configuration["ResourceServiceClient:ClientCertificate:Subject"];
-
-                if (subject is null or [])
-                {
-                    throw new InvalidOperationException("ResourceServiceClient:ClientCertificate:Source is \"KeyStore\", but no Certificate:FilePath is configured.");
-                }
+                var subject = _dashboardOptions.ResourceServiceClient.ClientCertificates.Subject!;
```

Comment:
Subject is never null?

Reply:
Validation will fail if it's null. Won't get to this point. I'll add a comment.